Foundation/NSNotification.swift

```diff
@@ -57,32 +58,43 @@ private let _defaultCenter: NSNotificationCenter = NSNotificationCenter()
 
 public class NSNotificationCenter : NSObject {
     
-    private var observers: [NSNotificationReceiver]
-    
-    public required override init() {
-        observers = [NSNotificationReceiver]()
-    }
+    private var _observers = [NSNotificationReceiver]()
+    private let _observersLock = NSLock()
     
     public class func defaultCenter() -> NSNotificationCenter {
         return _defaultCenter
     }
     
     public func postNotification(notification: NSNotification) {
+
         let name = notification.name
         let sender = notification.object
-        
-        let sendTo = observers.filter { observer in
+
+        _observersLock.lock()
+        let sendTo = _observers.filter { observer in
             let sameName = (observer.name == nil || observer.name == name)
             let sameSender = (observer.sender == nil || observer.sender === sender)
             
             return sameSender && sameName
         }
```

Comment:
In case of exception being thrown, the lines above lead to deadlock. The `defer` corresponds to the current scope. As an option, you can do smth like this:

```
var sendTo: [NSNotificationReceiver]!
{
	_observersLock.lock()
	defer {_observersLock.unlock()}
	sendTo = _observers.filter { ... }
}
```

but I would propose to define `synchronized` method and use it all over the class implementation:

```
private func synchronized<T>(lock: AnyObject, closure: () -> T) -> T {
    lock.lock()
    defer { lock.unlock() }
    return closure()
}

let sendTo = synchronized(_observersLock, {
	return _observers.filter { ... }
})
```